### PR TITLE
Adding test infrastructure to skycfg

### PR DESCRIPTION
**Pull request by choo-stripe**
We need a way of executing tests in skycfg, beyond just running `main`. Here's an implementation, open to suggestions.

### CODE REVIEW

File: skycfg.go [hunk 1]
```diff
@@ -215,12 +219,21 @@ func loadImpl(ctx context.Context, opts *loadOptions, filename string) (starlark
 		cache[modulePath] = nil
 		globals, err := starlark.ExecFile(thread, modulePath, moduleSource, opts.globals)
 		cache[modulePath] = &cacheEntry{globals, err}
+
+		for name, val := range globals {
+			if strings.HasPrefix(name, "test_") && val.Type() == "function" {
```
Comment (jmillikin-stripe) [inline skycfg.go]: I think we'll want to use a Go type cast here, so we don't need to make assumptions about Starlark type names.

```go
if !strings.HasPrefix(name, "test_") {
    continue
}
if fn, ok := val.(starlark.Callable); ok {
    tests = append(tests, &Test{
        callable: fn,
    })
}

```

File: skycfg.go [hunk 2]
```diff
@@ -315,6 +328,68 @@ func (c *Config) Main(ctx context.Context, opts ...ExecOption) ([]proto.Message,
 	return msgs, nil
 }
 
+// A TestResult is the result of a test run
+type TestResult struct {
+	Name     string
```
Comment (jmillikin-stripe) [inline skycfg.go]: nit: `Name` -> `TestName`

File: skycfg.go [hunk 3]
```diff
@@ -328,3 +403,37 @@ func skyFail(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwar
 	t.Caller().WriteBacktrace(&buf)
 	return nil, fmt.Errorf("[%s] %s\n%s", t.Caller().Position(), msg, buf.String())
 }
+
+// AssertionError represents a failed assertion
+type AssertionError struct {
```
Comment (jmillikin-stripe) [inline skycfg.go]: Lets name this `assertionError` so it doesn't clutter up in the godoc output.

File: skycfg.go (hunk 1 above)
Comment (choo-stripe) [reply skycfg.go]: Good call.

**Comment by choo-stripe** [comment]
@jmillikin-stripe updated :)

**Pull request merged**